kong/runloop/wasm.lua

```diff
@@ -920,12 +920,15 @@ function _M.attach(ctx)
     return
   end
 
-  ctx.ran_wasm = true
+  local ok, err
+  if not ctx.ran_wasm then
```

Comment:
@hishamhm can you tell me if this looks right?

Also, should I move `proxy_wasm.set_host_properties_handlers()` into this block, or is it expected to be called every time `proxy_wasm.start()` is called?

Reply:
I'm looking at it, and trying to understand what's the code path that leads to this being called again when a filter issues dispatch_http_call... 

Because if I'm following correctly, that's the sequence, right?

kong.runloop.wasm.attach()
called by kong.runloop.handler.wasm_attach()
called by kong.init.access()

...and that's the only place I see it happening, so a dispatched HTTP call triggers the Lua handler for the `access` phase again? I'm confused.

If we're trying to avoid the Lua side from setting up proxy-wasm again when it is already running, should the test perhaps be something like `if ctx.ran_wasm then return end` at the very top? I really don't know — do we have a way to check this behavior at least manually?

Reply:
I'm getting closer to a reliable repro, but there are still so many unknowns.

I _did_ manage to figure out that `proxy_wasm.set_host_properties_handlers()` should not be called more than once per request though, because after guarding against `attach()`, `set_host_properties_handlers()` returned an error to me about "property handlers already set."